sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java

```diff
@@ -149,6 +150,15 @@ public Mono<PollResponse<T>> onInitialResponse(Response<?> response, PollingCont
             Duration retryAfter = ImplUtils.getRetryAfterFromHeaders(response.getHeaders(), OffsetDateTime::now);
             return PollingUtils.convertResponse(response.getValue(), serializer, pollResponseType)
                 .map(value -> new PollResponse<>(LongRunningOperationStatus.IN_PROGRESS, value, retryAfter))
+                .onErrorResume(error ->
+                    error instanceof UncheckedIOException,
```

Comment:
does `UncheckedIOException` always mean deserialize error in this case?

Reply:
It may not always mean deserialize error. However I don't have more detailed exception from impl of `DefaultJsonSerializer`.

I can try test the `error.getCause` on `JsonProcessingException`, but this would be implementation specific, and won't work if the underlying `ObjectSerializer` impl changes.